libvncclient/sockets.h

```diff
@@ -0,0 +1,31 @@
+#ifndef SOCKETS_H
+#define SOCKETS_H
+
+/*
+ *  Copyright (C) 2020 Tobias Junghans <tobydox@veyon.io>
+ *
+ *  This is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This software is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this software; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ *  USA.
+ */
+
+#include <rfb/rfbproto.h>
+
+extern rfbSocket ConnectClientToTcpAddrWithTimeout(unsigned int host, int port, unsigned int timeout);
```

Comment:
why not add to public API?

Reply:
As discussed recently I vote for making essential functions part of the API only. This gives us greater flexibility for changes. We can start doing so by not adding new functions unncessarily. libvncclient IMHO does not have to expose the low-level network/socket layer since it's purpose is to provide an abstract/high-level VNC client. If you disagree, I can make the proposed changes though.